### PR TITLE
Add support for Tekton Triggers to the Tekton extension, fixes #2043

### LINKED ISSUE #2043: Include Tekton Trigger into the Tekton extension
The Tekton extension only supports Tekton Pipeline. We should add support for Tekton Triggers as well

### PR DESCRIPTION
**Pull request by Fabian-K**
## Description
Add support for Tekton Triggers to the Tekton extension, fixes #2043

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [x] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [x] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Review by manusa — APPROVED** [review]

**Review by iocanel — APPROVED** [review]
+1

**Review by rohanKanojia — APPROVED** [review]

File: extensions/tekton/model-triggers/src/main/java/io/fabric8/tekton/triggers/internal/pipeline/pkg/apis/pipeline/v1beta1/ArrayOrString.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+import java.util.List;
+import java.util.Map;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonDeserialize(using = ArrayOrString.Deserializer.class)
+@JsonSerialize(using = ArrayOrString.Serializer.class)
+@ToString
+@EqualsAndHashCode
+@Buildable(editableEnabled = false, validationEnabled = false, generateBuilderPackage = false, builderPackage = "io.fabric8.kubernetes.api.builder", inline = @Inline(type = Doneable.class, prefix = "Doneable", value = "done"))
+public class ArrayOrString implements KubernetesResource {
```
Comment (rohanKanojia) [inline extensions/tekton/model-triggers/src/main/java/io/fabric8/tekton/triggers/internal/pipeline/pkg/apis/pipeline/v1beta1/ArrayOrString.java:48]: I see this class is duplicated for each tekton model module. Do you think we can avoid this somehow by maybe adding tekton-model as dependency? Or Are triggers independent of both v1alpha1 or v1beta1 models?

**Comment by rohanKanojia** [comment]
@Fabian-K : Thanks a lot for this :heart: . Could you please resolve this minor conflict? I think we can proceed to merge. Do you think it's a good idea adding some example/doc of Tekton triggers usage?

File: extensions/tekton/model-triggers/src/main/java/io/fabric8/tekton/triggers/internal/pipeline/pkg/apis/pipeline/v1beta1/ArrayOrString.java (hunk 1 above)
Comment (Fabian-K) [reply extensions/tekton/model-triggers/src/main/java/io/fabric8/tekton/triggers/internal/pipeline/pkg/apis/pipeline/v1beta1/ArrayOrString.java]: Good Point! I replaced it with a v1beta1 model dependency

**Comment by Fabian-K** [comment]
I guess you can never have too much documentation... For now, I at least added a note to the cheatsheet as a starting point

**Comment by rohanKanojia** [comment]
[merge]

**Pull request merged**